app/server/command-line.js

```diff
@@ -2,75 +2,102 @@ const fs = require('fs');
 
 function checkCommandLine() {
-  const ejectIndex = process.argv.indexOf('eject');
-  if (ejectIndex !== -1) {
-    return {
-      commandName: 'eject',
-      args: process.argv.slice(ejectIndex + 1)
-    };
-  }
-
-  return null;
+  const cliArgs = yargs(hideBin(process.argv))
+    .command({
+      command: 'eject',
+      describe: 'Export one or multiple AnaVis files to JSON or CSV',
+      builder: yargs => yargs
+        .option('directory', { alias: 'd', demandOption: false, type: 'string' })
+        .option('format', { alias: 'f', demandOption: true, type: 'string', default: 'json' }),
+      handler: argv => {
+        argv.command = 'eject';
+        argv.args = {
+          directory: argv.directory,
+          format: argv.format === 'csv' ? 'csv' : 'json',
+          files: argv._.filter(arg => arg !== 'eject')
+        }
+      }
+    })
+    .help()
+    .argv;
+
+    return cliArgs.command ? cliArgs : null;
 }
 
-function runCommand({ commandName, args }, cb) {
-  switch (commandName) {
+function runCommand(cliArgs, cb) {
+  switch (cliArgs.command) {
     case 'eject':
-      doEject(args, cb);
+      doEject(cliArgs.args, cb);
       break;
     default:
-      cb(new Error(`Invalid command name: ${commandName}`));
+      cb(new Error(`Invalid command name: ${cliArgs.command}`));
       break;
   }
 }
 
 function doEject(args, cb) {
-  if (!args.length) return cb(new Error('No file name specified'));
-
-  const directoryMode = args.some(arg => ['-d', '--dir'].includes(arg));
-  const fileOrDirName = args.filter(arg => !['-d', '--dir'].includes(arg))[0];
-
-  if (!fileOrDirName) {
-    return cb(new Error(`Missing ${directoryMode ? 'directory' : 'file'} name`));
+  if (!args.directory && !args.files) {
+    return cb(new Error('No file name specified'));
   }
 
-  const collectFiles = directoryMode
-    ? func => glob(path.join(fileOrDirName, '*.avd'), func)
-    : func => func(null, [fileOrDirName]);
+  const collectFiles = args.directory
+    ? func => glob(path.join(args.directory, '*.avd'), func)
+    : func => func(null, args.files);
 
   collectFiles((err, files) => {
     if (err) cb(err);
 
     if (!files || !files.length) {
       // eslint-disable-next-line no-console
-      console.log(`No AnaVis files found in directory ${fileOrDirName}`);
+      console.log(`No AnaVis files found in directory ${args.directory}`);
     }
 
-    ejectAvdFiles(files, 0, cb);
+    ejectAvdFiles(files, 0, args.format, cb);
   });
 }
 
-function ejectAvdFiles(avdFileNames, index, cb) {
+function doc2Csv(doc) {
+  const sumPartLengths = parts => parts.reduce((accu, part) => accu + part.length, 0);
+
+  const totalLength = sumPartLengths(doc.parts);
+
+  const segments = doc.parts.map((part, index) => ({
+    startPosition: sumPartLengths(doc.parts.slice(0, index)) / totalLength,
+    title: part.name,
+    color: part.color,
+    text: doc.annotations.map(anno => anno.values[index]).join('\n\n') || ''
```

Comment:
Might be good to filter out empty annotations, or is that not possible to have in there?

Reply:
True, good idea! Done.